src/Trace/TraceClient.php

```diff
@@ -0,0 +1,199 @@
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::FULL_CONTROL_SCOPE];
+        }
+
+        $this->connection = new Rest($this->configureAuthentication($config));
+    }
+
+    /**
+     * Returns the projectId for this client.
+     *
+     * @return string
+     */
+    public function projectId()
+    {
+        return $this->projectId;
+    }
+
+    /**
+     * Sends a Trace log in a simple fashion.
+     *
+     * @param  Trace $trace The trace log to send.
+     * @return bool
+     * @throws ServiceException
+     */
+    public function insertTrace(Trace $trace)
+    {
+        return $this->insertTraceBatch([$trace]);
+    }
+
+    /**
+     * Sends multiple Trace logs in a simple fashion.
+     *
+     * @param Trace[] $traces The trace logs to send.
+     * @return bool
+     * @throws ServiceException
+     */
+    public function insertTraceBatch(array $traces)
+    {
+        // throws ServiceException on failure
+        $this->connection->patchTraces([
+            'projectId' => $this->projectId,
+            'traces' => array_map(function ($trace) {
+                return $trace->info();
+            }, $traces)
+        ]);
+        return true;
+    }
+
+    /**
+     * Fetch a single trace by traceId.
+     *
+     * @param  string $traceId The ID of the trace to return
+     * @return Trace
+     * @throws ServiceException
+     */
+    public function getTrace($traceId)
+    {
+        $trace = $this->connection->getTrace([
+            'projectId' => $this->projectId,
+            'traceId' => $traceId
+        ]);
+
+        if (empty($trace)) {
+            throw new NotFoundException('Trace ID does not exist', 404);
+        };
+
+        return new Trace($this->pluck('projectId', $trace), $trace);
+    }
+
+    /**
+     * Lazily find or instantiates a trace. There are no network requests made at this
+     * point. To see the operations that can be performed on a trace please
+     * see {@see Google\Cloud\Trace\Trace}.
+
+     * @param  string $traceId [optional] The trace id of the trace to reference.
+     * @return Trace
+     */
+    public function trace($traceId = null)
+    {
+        return new Trace($this->projectId, ['traceId' => $traceId]);
+    }
+
+    /**
+     * Fetch all traces in the project
+     *
+     * @param array $options [optional] {
+     *      Configuration options.
+     *
+     *      @type string $viewType Type of data returned for traces in the list.
+     *            Can be one of 'VIEW_TYPE_UNSPECIFIED', 'MINIMAL', 'ROOTSPAN', or
+     *            'COMPLETE'
+     *      @type integer $pageSize Maximum number of traces to return
+     *      @type string $pageToken Token identifying the page of results to return
+     *      @type string $startTime Start of the time interval during which trace data
```

Comment:
`$startTime`, `$endTime` and `$filter` could use a little extra detail, maybe an example or a relevant link?